tests/helpers/bonafide_helper.rb

```diff
@@ -17,58 +17,53 @@ def assert_tmp_user
     raise exc
   end
 
-  def api_url(path)
-    api = property('api')
-    "https://%{domain}:%{port}#{path}" % {
-      :domain   => api['domain'],
-      :port     => api['port']
-    }
-  end
-
   #
   # attempts to create a user account via the API,
   # returning the user object if successful.
   #
-  def assert_create_user
-    user = SRP::User.new
-    url = api_url("/1/users.json")
-
+  def assert_create_user(username=nil, auth=nil)
+    user = SRP::User.new(username)
+    url = api_url("/users.json")
     params = user.to_params
-
-    if property('webapp.invite_required')
-      @invite_code = generate_invite_code
-      params['user[invite_code]'] = @invite_code
+    if auth
```

Comment:
Is there a reason for nesting the generate_invite_code in this if clause?
It makes tests on mx fail because they use auth: :monitor and still require an invite token.
I think there are different ways around this...
* do not require invite codes if monitoring auth token is there
* do not use monitoring auth token here
* make sure param gets set no matter if monitor auth is used or not.

Not sure which one is appropriate. @elijh  ?